### PR TITLE
Milestone 1: Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by skabeo**
In this pull request

- The person class was created in `src/person.rb` file
- Teacher class was created as inheritance from person class in `src/teacher.rb`
- Student class was created as an inheritance from the Person class in `src/student.rb`

### CODE REVIEW

**Review by TedLivist — COMMENTED** [review]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @skabeo,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You have submitted a project that does not have descriptive README.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './person'
+# Student class demonstrating the inheritance principle
+class Student < Person
+  def initialize(classrooms, name = 'Unknown', age = 0, parent_permission: true)
+    super(name, age, parent_permission)
```
Comment (ShadyShawkat) [inline src/student.rb]: - [ ] Please make sure to pass the parameters in the right order to the `super` constructor. The `Person` constructor accepts the parameters in the following order:

```ruby
def initialize(age, name = 'Unknown', parent_permission: true)
```

- On the other hand, you're passing the parameters to it from the `super` constructor: this way:

```ruby
super(name, age, parent_permission)
```

File: src/teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './person'
+# Teacher class demonstrating the inheritance principle
+class Teacher < Person
+  def initialize(specialization, name = 'Unknown', age = 0, parent_permission: true)
+    super(name, age, parent_permission)
```
Comment (ShadyShawkat) [inline src/teacher.rb]: - [ ] Please make sure to pass the parameters in the right order to the `super` constructor. The `Person` constructor accepts the parameters in the following order:

```ruby
def initialize(age, name = 'Unknown', parent_permission: true)
```

- On the other hand, you're passing the parameters to it from the `super` constructor: this way:

```ruby
super(name, age, parent_permission)
```

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @skabeo, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/teacher.rb (hunk 2 above)
Comment (skabeo) [reply src/teacher.rb]: Hello @ShadyShawkat 

Thank for reviewing my code. A quick question how do i go around this if rubocop always flags an error for using the right order.

**Review by brainconnect93 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @skabeo, 👋 👋  

Your project is complete! Great job for following the project requirement🥇and implementing the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 




## Good Points 👍

- Descriptive Pull Request title & summary. ✔️
- Create an individual `.rb` files for each class (`Person`, `Student`, and `Teacher` ). ✔️
- Add Person, Student, and Teacher classes ✔️
- Good use of Git flow ✔️

## Optional changes

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brainconnect93 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**